tests/end-to-end/external-solution-integration/testkit/testservice.go

```diff
@@ -0,0 +1,221 @@
+package testkit
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/resourceskit"
+	log "github.com/sirupsen/logrus"
+	model "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	testServiceName       = "counter-service"
+	testServicePort       = 8090
+	testServiceImage      = "maladie/counterservice:latest"
+	labelKey              = "component"
+	healthEndpointFormat  = "http://%s.%s:%v/health"
+	counterEndpointFormat = "http://%s.%s:%v"
+)
+
+type TestService interface {
+	CreateTestService() error
+	DeleteTestService() error
+	checkValue() (int, error)
+	IsReady() (bool, error)
+	GetTestServiceURL() string
+	WaitForCounterPodToUpdateValue() (bool, error)
+}
+
+type testService struct {
+	K8sResourcesClient resourceskit.K8sResourcesClient
+	HttpClient         http.Client
+}
+
+func NewTestService(k8sClient resourceskit.K8sResourcesClient) TestService {
+
+	httpClient := newHttpClient(true)
+
+	return &testService{
+		K8sResourcesClient: k8sClient,
+		HttpClient:         *httpClient,
+	}
+}
+
+func (ts *testService) CreateTestService() error {
+	err := ts.createDeployment()
+	if err != nil {
+		return err
+	}
+	err = ts.createService()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ts *testService) checkValue() (int, error) {
+
+	url := ts.GetTestServiceURL() + "/counter"
+
+	resp, err := ts.HttpClient.Get(url)
+
+	if err != nil {
+		return 0, err
+	}
+
+	defer resp.Body.Close()
+
+	var response struct {
+		Counter int `json:"counter"`
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&response)
+
+	if err != nil {
+		return 0, err
+	}
+
+	return response.Counter, nil
+}
+
+func (ts *testService) IsReady() (bool, error) {
+
+	url := ts.getHealthEndpointURL()
+	log.WithFields(log.Fields{"url": url}).Debug("IsReady?")
+
+	resp, _ := ts.HttpClient.Get(url)
+
+	// lets ignore all errors here, shall we?
+
+	// if err != nil {
+	// return false, err
+	// }
+	if resp != nil {
+		if resp.StatusCode == http.StatusOK {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+func (ts *testService) WaitForCounterPodToUpdateValue() (bool, error) {
+	count, err := ts.checkValue()
+	if err != nil {
+		log.Error(err)
+		return false, err
+	}
+
+	if count != 1 {
```

Comment:
The better approach would be to pass expected count as int in the arguments, this way we would have:
`func (ts *testService) WaitForCounterPodToUpdateValue(expected int) (bool, error)`

We could later set a constant to provide expected CounterService value, what do you think?

Reply:
fixed.